src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -790,6 +798,8 @@ void grpc_chttp2_incoming_byte_stream_finished(
 void grpc_chttp2_ack_ping(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
                           uint64_t id);
 
+void grpc_chttp2_ping_strike(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t);
```

Comment:
Suggest renaming this function. Not exactly sure what - but maybe something like `grpc_chttp2_check_max_ping_strikes` and adding a comment that this sends a go-away frame if num pings > max-ping-strikes